### PR TITLE
Add translation

### PR DESCRIPTION
**Pull request by GregoireHebert**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #40
| License       | MIT

### CODE REVIEW

File: src/Translator/TranslatorAware.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace DocusignBundle\Translator;
+
+use Symfony\Component\Translation\TranslatorInterface;
+
+interface TranslatorAware
```
Comment (vincentchalamon) [inline src/Translator/TranslatorAware.php]:
```suggestion
interface TranslatorAwareInterface
```

File: src/Translator/TranslatorTrait.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace DocusignBundle\Translator;
+
+use Symfony\Component\Translation\IdentityTranslator;
+use Symfony\Component\Translation\TranslatorInterface;
+
+trait TranslatorTrait
```
Comment (vincentchalamon) [inline src/Translator/TranslatorTrait.php]:
```suggestion
trait TranslatorAwareTrait
```

File: src/Controller/Callback.php [hunk 3]
```diff
@@ -13,22 +13,39 @@
...
     public const EVENT_COMPLETE = 'signing_complete';
 
+    use TranslatorTrait;
+
     public function __invoke(Request $request, EventDispatcherInterface $eventDispatcher): Response
     {
         if (self::EVENT_COMPLETE !== $status = $request->get('event')) {
-            return new Response("The document signature ended with an unexpected $status status.");
+            return new Response(
+                $this->translator->trans(
```
Comment (vincentchalamon) [inline src/Controller/Callback.php]:
```suggestion
                $this->getTranslator->()trans(
```

Comment (vincentchalamon) [inline src/Controller/Callback.php]:
```suggestion
                $this->getTranslator()->trans(
```

File: src/DependencyInjection/Compiler/TranslationPass.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace DocusignBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\Translation\TranslatorInterface;
+
+final class TranslationPass implements CompilerPassInterface
```
Comment (vincentchalamon) [inline src/DependencyInjection/Compiler/TranslationPass.php]:
```suggestion
final class TranslationCompilerPass implements CompilerPassInterface
```

File: src/Controller/Callback.php [hunk 5]
```diff
@@ -13,22 +13,39 @@
...
+                    'The document signature ended with an unexpected %status% status.',
+                    ['%status%' => $status],
+                    DocusignBundle::TRANSLATION_DOMAIN
+                )
+            );
         }
 
-        $event = new DocumentSignatureCompletedEvent($request, new Response('Congratulation! Document signed.'));
+        $event = new DocumentSignatureCompletedEvent($request, new Response(
+            $this->translator->trans(
```
Comment (vincentchalamon) [inline src/Controller/Callback.php]:
```suggestion
            $this->getTranslator()->trans(
```

File: src/DocusignBundle.php [hunk 6]
```diff
@@ -20,6 +20,8 @@
 
 final class DocusignBundle extends Bundle
 {
+    public const TRANSLATION_DOMAIN = 'docusign_bundle';
```
Comment (vincentchalamon) [inline src/DocusignBundle.php]: Should it be moved to `TranslatorAwareInterface` instead?

File: src/EnvelopeCreator/DefineEnvelope.php [hunk 7]
```diff
@@ -39,7 +47,7 @@ public function __invoke(array $context = []): void
         }
 
         $this->envelopeBuilder->setEnvelopeDefinition(new Model\EnvelopeDefinition([
-            'email_subject' => self::EMAIL_SUBJECT,
+            'email_subject' => $this->translator->trans(self::EMAIL_SUBJECT, [], DocusignBundle::TRANSLATION_DOMAIN),
```
Comment (vincentchalamon) [inline src/EnvelopeCreator/DefineEnvelope.php]:
```suggestion
            'email_subject' => $this->getTranslator()->trans(self::EMAIL_SUBJECT, [], DocusignBundle::TRANSLATION_DOMAIN),
```

File: src/Resources/translations/docusign_bundle.de.xlf [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+            <trans-unit id="docusign.sign_this">
+                <source>Please sign this document.</source>
+                <target>Bitte unterschreiben Sie dieses Dokument.</target>
+            </trans-unit>
+            <trans-unit id="docusign.status">
+                <source>The document signature has ended with an unexpected status "%status%".</source>
+                <target>Die Dokumentensignatur wurde mit einem unerwarteten Status "%status%" beendet.</target>
+            </trans-unit>
+            <trans-unit id="docusign.document_signed">
+                <source>Congratulation! Document signed.</source>
```
Comment (vincentchalamon) [inline src/Resources/translations/docusign_bundle.de.xlf]:
```suggestion
                <source>Congratulations! The document have been successfully signed.</source>
```
Also to apply to other translations

File: src/DependencyInjection/Compiler/TranslationPass.php (hunk 4 above)
Comment (GregoireHebert) [reply src/DependencyInjection/Compiler/TranslationPass.php]: then I should use the same for pluginpass

File: src/DocusignBundle.php (hunk 6 above)
Comment (GregoireHebert) [reply src/DocusignBundle.php]: At the moment I feel it like it's the right place...
When overriding from a symfony project perspective, it kinda makes sense that it come from the bundle.

File: src/EnvelopeCreator/DefineEnvelope.php [hunk 9]
```diff
@@ -14,22 +14,30 @@
...
     private $router;
     private $envelopeBuilder;
+    private $translator;
 
-    public function __construct(EnvelopeBuilderInterface $envelopeBuilder, RouterInterface $router)
+    public function __construct(EnvelopeBuilderInterface $envelopeBuilder, RouterInterface $router, TranslatorInterface $translator)
     {
         $this->router = $router;
         $this->envelopeBuilder = $envelopeBuilder;
+        $this->translator = $translator;
```
Comment (vincentchalamon) [inline src/EnvelopeCreator/DefineEnvelope.php]: Duplicate from TranslatorAwareTrait

File: src/EnvelopeCreator/DefineEnvelope.php [hunk 10]
```diff
@@ -14,17 +14,23 @@
...
+final class DefineEnvelope implements EnvelopeBuilderCallableInterface, TranslatorAwareInterface
 {
     public const EMAIL_SUBJECT = 'Please sign this document';
     public const WEBHOOK_ROUTE_NAME = 'docusign_webhook';
 
+    use TranslatorAwareTrait;
+
     private $router;
     private $envelopeBuilder;
+    private $translator;
```
Comment (vincentchalamon) [inline src/EnvelopeCreator/DefineEnvelope.php]: Duplicate from trait

Comment (GregoireHebert) [reply src/EnvelopeCreator/DefineEnvelope.php]: f**k xD

**Pull request merged**